### PR TITLE
fix(sdcm/tester.py): Check if version_str is non-empty

### PR DESCRIPTION
**Pull request by k0machi**
This corrects an issue where argus_get_scylla_version would fail if BaseNode.get_scylla_binary_version() returns None. In this case it logs a warning instead.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by fruch — DISMISSED** [review]
LGTM

**Review by fgelcer — DISMISSED** [review]
LGTM

File: sdcm/tester.py [hunk 1]
```diff
@@ -426,9 +426,11 @@ def argus_get_scylla_version(self):
             self.log.info("Collection Scylla version for argus...")
             version_regex = re.compile(r'(([\w.~]+)-(0.)?([0-9]{8,8}).(\w+).)')
             version_str = self.db_cluster.nodes[0].get_scylla_binary_version()
-            if (match := version_regex.match(version_str)):
+            if version_str and (match := version_regex.match(version_str)):
                 self.argus_test_run.run_info.details.scylla_version = match.group(2)
                 self.argus_test_run.save()
+            else:
```
Comment (roydahan) [inline sdcm/tester.py:432]: Why only warning?
I think we want to fail the test and check why we couldn’t get the version.

**Review by roydahan — CHANGES_REQUESTED** [review]

File: sdcm/tester.py (hunk 1 above)
Comment (k0machi) [reply sdcm/tester.py]: Done

File: sdcm/tester.py [hunk 2]
```diff
@@ -426,11 +426,13 @@ def argus_get_scylla_version(self):
             self.log.info("Collection Scylla version for argus...")
             version_regex = re.compile(r'(([\w.~]+)-(0.)?([0-9]{8,8}).(\w+).)')
             version_str = self.db_cluster.nodes[0].get_scylla_binary_version()
-            if (match := version_regex.match(version_str)):
+            if version_str and (match := version_regex.match(version_str)):
                 self.argus_test_run.run_info.details.scylla_version = match.group(2)
                 self.argus_test_run.save()
+            return
```
Comment (ShlomiBalalis) [inline sdcm/tester.py:432]: Shouldn't the `return` be inside the if?

File: sdcm/tester.py [hunk 3]
```diff
@@ -426,11 +426,13 @@ def argus_get_scylla_version(self):
...
         except Exception:  # pylint: disable=broad-except
             self.log.error("Error getting scylla version for argus", exc_info=True)
+        raise CriticalTestFailure("Failed to get scylla version for argus")
```
Comment (roydahan) [inline sdcm/tester.py:435]: Don't we have Error? why critical?

File: sdcm/tester.py (hunk 2 above)
Comment (k0machi) [reply sdcm/tester.py:432]: Fixed

File: sdcm/tester.py (hunk 3 above)
Comment (k0machi) [reply sdcm/tester.py]: I've replaced it with just `Exception` for now, don't see other classes used for this in tester, but I see that unit tests are failing now, so maybe will have to patch this function out of the unit tests.

**Review by ShlomiBalalis — DISMISSED** [review]

File: sdcm/tester.py (hunk 3 above)
Comment (fruch) [reply sdcm/tester.py]: we should raise an event here, if we don't want to stop the test.

any exception here would stop the test (regardless of it's name), it's part of the test setup, and if it fails the test would won't run.

**Comment by roydahan** [comment]
It looks better now, although I don't really understand why it needs such a special care.

@fruch can you please review and merge if approve?

**Review by roydahan — APPROVED** [review]

**Review by fgelcer — APPROVED** [review]
LGTM

**Comment by roydahan** [comment]
@fgelcer Let's make sure it works on master and then backport.

**Comment by fgelcer** [comment]
> @fgelcer Let's make sure it works on master and then backport.

@roydahan , [last run](https://jenkins.scylladb.com/job/scylla-master/job/artifacts-offline-install/job/artifacts-ubuntu2004-nonroot-test/441/) passed, but anyway, last few runs also passed, even without this fix (probably there was a change in Scylla master, or other change in SCT), but it did not fail...

**Comment by roydahan** [comment]
IIRC, the test should FAIL with this PR merged.
@k0machi please check the latest run and check if it behaves correctly.

**Comment by k0machi** [comment]
> IIRC, the test should FAIL with this PR merged.
> @k0machi please check the latest run and check if it behaves correctly.

No runs yet with this fix, so we'll have to wait until next week trigger I think.

**Comment by fgelcer** [comment]
> > IIRC, the test should FAIL with this PR merged.
> > @k0machi please check the latest run and check if it behaves correctly.
> 
> No runs yet with this fix, so we'll have to wait until next week trigger I think.

yes, we had a run [in here](https://jenkins.scylladb.com/job/scylla-master/job/artifacts-offline-install/job/artifacts-ubuntu2004-test/490/)

**Comment by k0machi** [comment]
> > > IIRC, the test should FAIL with this PR merged.
> > > @k0machi please check the latest run and check if it behaves correctly.
> > 
> > No runs yet with this fix, so we'll have to wait until next week trigger I think.
> 
> yes, we had a run [in here](https://jenkins.scylladb.com/job/scylla-master/job/artifacts-offline-install/job/artifacts-ubuntu2004-test/490/)

This one is from September 19th, not October

**Comment by fgelcer** [comment]
> This one is from September 19th, not October

ah, you are right... i saw the 19th and thought it was today... but i had run one manually.. let me find it

**Comment by fgelcer** [comment]
@k0machi , so [here it is](https://jenkins.scylladb.com/job/scylla-master/job/artifacts-offline-install/job/artifacts-ubuntu2004-nonroot-test/442/)
see that [in Argus](https://argus.scylladb.com/workspace?state=WyJkN2U0N2M2My0wMzgwLTQ0YjQtYmQyNS02ZWJjNjEzMjE3ODUiXQ) we don't have the version reported, it failed with:
```
Unable to collect package versions for Argus - skipping...
Traceback (most recent call last):
  File "/home/jenkins/slave/workspace/scylla-master/artifacts-offline-install/artifacts-ubuntu2004-nonroot-test/scylla-cluster-tests/sdcm/tester.py", line 419, in argus_collect_packages
    self.argus_test_run.run_info.details.packages.append(self.generate_scylla_server_package())
  File "/home/jenkins/slave/workspace/scylla-master/artifacts-offline-install/artifacts-ubuntu2004-nonroot-test/scylla-cluster-tests/sdcm/tester.py", line 394, in generate_scylla_server_package
    version_dict = expr.match(scylla_version).groupdict()
AttributeError: 'NoneType' object has no attribute 'groupdict'
Collection Scylla version for argus...
```

the test itself seems to have passed (although there were errors during the test phase), but all the "steps" of the test seem to be completed successfully

**Comment by fruch** [comment]
> @k0machi , so [here it is](https://jenkins.scylladb.com/job/scylla-master/job/artifacts-offline-install/job/artifacts-ubuntu2004-nonroot-test/442/)
> see that [in Argus](https://argus.scylladb.com/workspace?state=WyJkN2U0N2M2My0wMzgwLTQ0YjQtYmQyNS02ZWJjNjEzMjE3ODUiXQ) we don't have the version reported, it failed with:
> ```
> Unable to collect package versions for Argus - skipping...
> Traceback (most recent call last):
>   File "/home/jenkins/slave/workspace/scylla-master/artifacts-offline-install/artifacts-ubuntu2004-nonroot-test/scylla-cluster-tests/sdcm/tester.py", line 419, in argus_collect_packages
>     self.argus_test_run.run_info.details.packages.append(self.generate_scylla_server_package())
>   File "/home/jenkins/slave/workspace/scylla-master/artifacts-offline-install/artifacts-ubuntu2004-nonroot-test/scylla-cluster-tests/sdcm/tester.py", line 394, in generate_scylla_server_package
>     version_dict = expr.match(scylla_version).groupdict()
> AttributeError: 'NoneType' object has no attribute 'groupdict'
> Collection Scylla version for argus...
> ```
> 
> the test itself seems to have passed (although there were errors during the test phase), but all the "steps" of the test seem to be completed successfully

The code in the PR is missing `.publish_or_dump()`, 

The event is sent, hence the test doesn't fail as you would expect.

We we are so keen to make that test fail, and not fixing the root cause ?

**Comment by fgelcer** [comment]
IMHO i agree we shouldn't fail the test, but find the root cause, and fix it (or report an issue to fix it), but we were experiencing false positives, as the tests were failing during setup, but the test was still being marked as success, and that is not ok

**Pull request merged**